symengine/polys.h

```diff
@@ -0,0 +1,70 @@
+/**
+ *  \file polys.h
+ *  Class for sparse Polynomial: Polynomial
+ *
+ **/
+#ifndef SYMENGINE_POLYS_H
+#define SYMENGINE_POLYS_H
+
+#include <symengine/basic.h>
+#include <symengine/dict.h>
+
+namespace SymEngine {
+//! Polynomial Class
+class Polynomial : public Basic{
+public:
+    //! `vars_` : Variables of the Polynomial
+    //! `polys_set_` : holds the Polynomial
+    vec_symbol vars_;
+    hash_set polys_set_;
```

Comment:
@certik You could use an ``std::unique_ptr`` to a ``hash_set``, instead of a straight ``hash_set``. Like this, it is not necessary to know the full definition of ``hash_set`` in the header. There's going to be one extra redirection due to the pointer, not sure how this would affect performance.